BUILD

```diff
@@ -39,6 +39,23 @@ java_library(
     ],
 )
 
+load("//tools:jarjar.bzl", "jarjar_library")
+
+genrule(
+    name = "rules_file",
+    outs = ["rules_file.txt"],
+    cmd = "echo \"rule com.google.auto.common.** dagger.shaded.auto.common.@1\" > $@",
+)
+
+jarjar_library(
+    name = "shaded_compiler",
+    rules_file = ":rules_file.txt",
+    deps = [
+        "//compiler",
+        "@com_google_auto_auto_common//jar",
```

Comment:
If you decide to continue shading javapoet please also relocate it. Other annotation processors using newer versions of javapoet might run into issues otherwise. For example https://github.com/rharter/auto-value-parcel/issues/64